### PR TITLE
Improvement: Adding scaling configuration according to processing mode.

### PR DESCRIPTION
**Pull request by FurongZhang**
The driver pipeline will be optimized to fix function SFC if power saving mode, media kernel path if performance mode.

Signed-off-by: furongzh <furong.zhang@intel.com>

### CODE REVIEW

File: media_driver/linux/common/vp/ddi/media_libva_vp.c [hunk 1]
```diff
@@ -1142,6 +1143,27 @@ DdiVp_SetProcPipelineParams(
...
+    switch (pPipelineParam->processing_mode)
+    {
+    case VAProcPowerSavingMode:
+        if (pVpHalSrcSurf->SurfType == SURF_IN_PRIMARY)
+        {
+            pVpHalSrcSurf->ScalingMode       = VPHAL_SCALING_AVS;
+            pVpHalSrcSurf->ScalingPreference = VPHAL_SCALING_PREFER_SFC;
+        }
+        break;
+    case VAProcPerformanceMode:
```
Comment (lizhong1008) [inline media_driver/linux/common/vp/ddi/media_libva_vp.c]: I doubt AVS can show better performance.  Probably using VAProcDefaultMode is better here? Any other thought?

Comment (FurongZhang) [reply media_driver/linux/common/vp/ddi/media_libva_vp.c]: Generally, AVS sampler has better performance for typical usage(such as scaling, CSC etc.). We have related performance data to prove that.

Comment (FurongZhang) [reply media_driver/linux/common/vp/ddi/media_libva_vp.c]: And if the VAAPI user does not set processing mode, we used the default mode(value = 0) which makes it compatible with the previous one.

**Comment by FurongZhang** [comment]
Longfei, can you merge it? Thanks  lot in advance!

**Pull request closed without merging**